instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/client/NettyNetClientAttributesExtractor.java

```diff
@@ -20,6 +20,11 @@
   public InetSocketAddress getAddress(
       HttpRequestAndChannel requestAndChannel, @Nullable HttpResponse response) {
     SocketAddress address = requestAndChannel.channel().remoteAddress();
+    // if the remote address is null at the end of request processing (because of e.g. lost
+    // connection) use the one captured when the operation started
+    if (address == null) {
+      address = requestAndChannel.remoteAddress();
+    }
```

Comment:
I don't think remoteAddress will change since netty instrumentation is tracing a single low-level request/response

```suggestion
    SocketAddress address = requestAndChannel.remoteAddress();
```